R/deleteThis.R

```diff
@@ -0,0 +1,1697 @@
+  #   the reusable items in the state as they were returned from the analysis
+  #
+  state <- .retrieveState()
+  if (! is.null(state)) {
+    if (! is.null(stateKey)) { # new method, from json
+      if ("baseSets" %in% names(stateKey)) # special placeholders will be used in state items
+        key <- .addBaseSetsState(stateKey)
+      else # no special placeholders in any set of options
+        key <- stateKey
+    } else if ("key" %in% names(attributes(state))) { # old method, from analysis
+      key <- attributes(state)$key
+    } else {
+      return(NULL)
+    }
+    state <- .getStateItems(state=state, options=options, key=key)
+  }
+  return(state)
+}
+
+.addBaseSetsState <- function(stateKey) {
+  # Replaces placeholders in state keys with the options associated with those
+  # placeholders
+  #
+  # Args:
+  # - stateKey: named list (from the analysis json) where each name corresponds to a
+  #             name in the state and the value to options in the options list.
+  #             the named entry 'baseSets' contains a list with placeholders that must be
+  #             replaced in the values of the other lists
+  #
+  # Return:
+  # - the altered state keys, now only containing options from the options list
+  #
+  baseSets <- stateKey[["baseSets"]]
+  stateKey[["baseSets"]] <- NULL
+  key <- lapply(stateKey, function(x) {
+    if (any(names(baseSets) %in% x)) {
+      sets <- baseSets[which(names(baseSets) %in% x)]
+      set <- unlist(sets, use.names=FALSE)
+      x <- x[which(! x %in% names(baseSets))]
+      return(c(x, set))
+    }
+    return(x)
+  })
+  return(key)
+}
+
+callback <- function(results=NULL, progress=NULL) {
+
+  ret <- 0
+
+  if (base::exists(".callbackNative")) {
+
+    if (is.null(results)) {
+      json.string <- "null"
+    } else {
+      json.string <- toJSON(.imgToResults(results))
+    }
+
+    if (is.null(progress)) {
+      progress <- -1
+    } else if (! is.numeric(progress)) {
+      stop("Provide a numeric value to the progress updater")
+    }
+
+    response <- .fromRCPP(".callbackNative", json.string, progress)
+
+    if (is.character(response)) {
+
+      ret <- fromJSON(base::paste("[", response, "]"))[[1]]
+
+    } else {
+
+      ret <- response
+    }
+  }
+
+  ret
+}
+
+.cat <- function(object) {
+
+  cat(toJSON(object))
+
+}
+
+.dataFrameToRowList <- function(df, discard.column.names=FALSE) {
+
+  if (dim(df)[1] == 0 || dim(df)[2] == 0)
+    return(list())
+
+  column.names <- names(df)
+  rows <- list()
+
+  for (i in 1:dim(df)[1]) {
+
+    row <- list()
+
+    for (j in 1:length(column.names))
+      row[[j]] <- df[i,j]
+
+    if ( ! discard.column.names)
+      names(row) <- column.names
+
+    rows[[i]] <- row
+  }
+
+  rows
+}
+
+.indices <- function(v) {
+
+  indices <- c()
+
+  if (length(v) > 0)
+    indices <- 1:length(v)
+
+  indices
+}
+
+.seqx <- function(from, to) {
+
+  if (from > to)
+    seq <- c()
+  else
+    seq <- from:to
+
+  seq
+}
```

Comment:
These are used in the ANOVA's

Reply:
I'll delete them there (especially considering the first one is just re-implementing `seq_along`)